pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/PulsarSource.java

```diff
@@ -145,9 +158,42 @@ public void close() throws Exception {
     @VisibleForTesting
     Map<String, ConsumerConfig<T>> setupConsumerConfigs() throws ClassNotFoundException {
         Map<String, ConsumerConfig<T>> configs = new TreeMap<>();
-
-        Class<?> typeArg = Reflections.loadClass(this.pulsarSourceConfig.getTypeClassName(),
-                Thread.currentThread().getContextClassLoader());
+        Class<?> typeArg;
```

Comment:
Is this change related to "Make Function submission checks optional"

Reply:
Can we do these checks in the JavaInstanceRunnable so that we don't complicate the logic here.  If the sourceSpec type classname is not set than and get the types from the jar so we can still get pass the source type.  I guess we also don't need to use a string to represent the type class, you can just pass in Class<?> typeClass in the pulsar source config